### PR TITLE
Adds new constants and unify Constants usage

### PR DESCRIPTION
**Pull request by Khaaz**
This PR does a general code cleanup:
- removes hardcoded values for `channel.type` and `message.type`
- uniformises Constants usage in the lib (always destructure constants).
It renames usage of OPCode to `GatewayOPCodes` or `VoiceOPCodes` and makes it more explicit.

*updates typings with new constants*

### CODE REVIEW

**Review by abalabahaha — APPROVED** [review]

File: lib/Constants.js [hunk 1]
```diff
@@ -172,3 +172,29 @@ module.exports.MessageFlags = {
...
+    DEFAULT: 0,
+    RECIPIENT_ADD: 1,
+    RECIPIENT_REMOVE: 2,
+    CALL: 3,
+    CHANNEL_NAME_CHANGE: 4,
+    CHANNEL_ICON_CHANGE: 5,
+    CHANNEL_PINNED_MESSAGE: 6,
+    GUILD_MEMBER_JOIN: 7,
+    USER_PREMIUM_GUILD_SUBSCRIPTION: 8,
+    USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_1:	9,
```
Comment (abalabahaha) [inline lib/Constants.js]: Hold on, are those tabs?

File: lib/Constants.js [hunk 2]
```diff
@@ -172,3 +172,29 @@ module.exports.MessageFlags = {
...
+    USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_2:	10,
```
Comment (abalabahaha) [inline lib/Constants.js]: Yeah that's a tab

File: lib/Constants.js [hunk 3]
```diff
@@ -172,3 +172,29 @@ module.exports.MessageFlags = {
...
+    USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_3:	11,
```
Comment (abalabahaha) [inline lib/Constants.js]: Use spaces

**Review by abalabahaha — CHANGES_REQUESTED** [review]

File: lib/Constants.js (hunk 1 above)
Comment (Khaaz) [reply lib/Constants.js]: nice catch, fixed

**Comment by Khaaz** [comment]
Also uses ChannelTypes in Shard.js.

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**